logstash-core-event-java/src/main/java/com/logstash/RubyJavaObject.java

```diff
@@ -0,0 +1,98 @@
+package com.logstash;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.jruby.Ruby;
+import org.jruby.RubyNil;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.io.Serializable;
+
+public class RubyJavaObject implements Serializable {
+    private IRubyObject rubyValue;
+    private Object javaValue;
+    private boolean is_ruby_nil;
+
+    public RubyJavaObject(Object javaValue) {
+        this.javaValue = javaValue;
+        this.rubyValue = null;
+    }
+
+    public RubyJavaObject(IRubyObject rubyValue) {
+        this.rubyValue = rubyValue;
+        is_ruby_nil = (this.rubyValue instanceof RubyNil);
+        this.javaValue = null;
+    }
+
+    public IRubyObject getRubyValue(Ruby runtime) {
+        if (rubyValue == null) {
+            rubyValue = Rubyfier.deep(runtime, javaValue);
+        }
+        return rubyValue;
+    }
+
+    @JsonValue
+    public Object getJavaValue() {
+        if (is_ruby_nil) {
+            return null;
+        }
+        if (javaValue == null) {
+            javaValue = Javafier.deep(rubyValue);
+        }
+        return javaValue;
+    }
+
+    public void update(final RubyJavaObject other) {
+        this.rubyValue = other.getRawRubyValue();
+        this.javaValue = other.getRawJavaValue();
+    }
+
+    protected IRubyObject getRawRubyValue() {
+        return rubyValue;
+    }
+
+    protected Object getRawJavaValue() {
+        return javaValue;
+    }
+
+    public Object javaGetRubyValue() {
+        return rubyValue;
+    }
+
+    public boolean hasRubyValue() {
+        return null != rubyValue;
+    }
+
+    public boolean hasJavaValue() {
+        return null != javaValue;
+    }
+
+    public boolean isRubyNil() {  return is_ruby_nil; }
+
+    @Override
+    public String toString() {
+        final StringBuffer sb = new StringBuffer("RubyJavaObject{");
```

Comment:
Probably makes sense to use a StringBuilder IMHO. Should be a little faster, but its not a big deal either way.